### ISSUE TITLE
Inconsistency in probed crtc list

### ISSUE THREAD

**Issue by vberger** [Smithay/drm-rs#21]
The crate seems to output a wrong list of CRTCs on my system. I used [this example program](https://gist.github.com/vberger/5335b90ecbe00a97c51a6989c0b8ff59) and compared its output to the output of the modetest tool from mesa (see [instructions for using it](https://wiki.linaro.org/xinliang/libdrm/modetest)), and they differ on the crtc list.

This is the list printed by the drm-rs example:

```
[
    (
        crtc::Handle(63),
        Err(Error(Unix(Sys(ENOENT)), State { next_error: None } ) )
    ),
    (
        crtc::Handle(64),
        Err(Error(Unix(Sys(ENOENT)), State { next_error: None } ) )
    ),
    (
        crtc::Handle(46),
        Ok(
            Info {
                handle: crtc::Handle(46),
                position: (0, 0),
                fb: framebuffer::Handle(0),
                gamma_length: 256
            }
        )
    )
]

```
while modetest lists crtcs with numbers `32`, `39` and `46`.

Note also that the reported `63` and `64` crtcs here seem to be invalid, as trying to query their info returns an error.

**Comment by Drakulix** [comment]
I tried to debug this by comparing `drm-rs` and `modetest`. `modetest` just calls `drmModeGetResources`, which is almost identical to `Device::resource_handles()`.

To compare:

- [libdrm source](https://cgit.freedesktop.org/mesa/drm/tree/xf86drmMode.c#n157)
- [drm-rs source](https://github.com/Smithay/drm-rs/blob/master/src/control/mod.rs#L120-L147)

My only possible explanation is that somehow `Handle(control::RawHandle)` is not the same size as a `u32`, but that was ruled out by @vberger by adding `[repr(C)]` to the `Handle` type.

I am out of ideas, hopefully @Slabity has some more.

**Comment by Slabity** [comment]
```rust
let ids = ResourceHandles {
    connectors: ffi_buf!(raw.connector_id_ptr, raw.count_connectors),
    encoders: ffi_buf!(raw.encoder_id_ptr, raw.count_connectors),
    crtcs: ffi_buf!(raw.crtc_id_ptr, raw.count_crtcs),
    framebuffers: ffi_buf!(raw.fb_id_ptr, raw.count_fbs),
    width: (raw.min_width, raw.max_width),
    height: (raw.min_height, raw.max_height)
};
```

We seem to be setting the size of the encoder buffer to `count_connectors` instead of `count_encoders`. My theory is that the driver is overflowing to the next buffer, overwriting some CRTC ids with encoder ids instead.

You can confirm this by trying to load `63` and `64` as encoders. But in any case, it's a bug. I'll patch it now.

**Comment by Slabity** [comment]
Fixed in 0f3a753

**Comment by vberger** [comment]
I can confirm this fixes the issue. 👍
